gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -694,13 +694,6 @@ MacroExpander::match_repetition (Parser<MacroInvocLexer> &parser,
       gcc_unreachable ();
     }
 
-  if (!res)
-    rust_error_at (rep.get_match_locus (),
-		   "invalid amount of matches for macro invocation. Expected "
-		   "between %s and %s, got %lu",
-		   lo_str.c_str (), hi_str.c_str (),
-		   (unsigned long) match_amount);
-
```

Comment:
The problem is that we are emitting the error *too* eagerly when another match arm could match the invocation, not that the error is emitted. We cannot just remove the error emission - we need to delay it until all match arms have been tried

Reply:
It does still fail though -- note the ```// { dg-error "Failed to match any rule within macro" }```. While the error is less descriptive, it is similar to the error in the linked rust playground code (```no rules expected the token '12'```) anyway.